action.yml

```diff
@@ -36,77 +40,66 @@ outputs:
-        branch_id=$(curl --silent \
-            "https://console.neon.tech/api/v2/projects/${PROJECT_ID}/branches" \
-            --header "Accept: application/json" \
-            --header "Content-Type: application/json" \
-            --header "Authorization: Bearer ${API_KEY}" \
-            | jq -r .branches \
-            | jq -c '.[] | select(.name | contains("'${BRANCH_NAME}'")) .id' \
-            | jq -r \
-            ) \
+        branch_id=$(yarn -s neonctl branches list --project.id ${{ inputs.project_id }} -o json \
+            | jq -c '.[] | select(.name | contains("'${{ inputs.branch_name }}'")) .id' \
+            | jq -r)
 
+        echo "branch exists, branch id: ${branch_id}" >> debug.log
 
-        # Get the list of endpoints for the branch by its id. 
-        endpoints=$(curl --silent \
-            "https://console.neon.tech/api/v2/projects/${PROJECT_ID}/branches/${branch_id}/endpoints" \
-            --header "Accept: application/json" \
-            --header "Content-Type: application/json" \
-            --header "Authorization: Bearer ${API_KEY}" \
-            ) \
+        # Get the list of endpoints for the branch by its id.
+        endpoints=$(yarn -s neonctl endpoints list \
+          --project.id ${{ inputs.project_id }} \
+          --branch.id $branch_id \
+          -o json)
 
-        endpoint_id=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f1)
-        region=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f2)
-        cloud_provider=$(echo $endpoints | jq --raw-output '.endpoints[0].host' | cut -d'.' -f3)
-        host=$(echo $endpoints | jq --raw-output '.endpoints[0].host')
-        branch_id=$(echo $endpoints | jq --raw-output '.endpoints[0].branch_id')
+        endpoint_id=$(echo $endpoints | jq --raw-output '.[0].id')
+        proxy_host=$(echo $endpoints | jq --raw-output '.[0].proxy_host')
+        host=$(echo $endpoints | jq --raw-output '.[0].host')
 
         echo "host=${host}" >> $GITHUB_OUTPUT
         echo "endpoint_id=${endpoint_id}" >> $GITHUB_OUTPUT
         echo "branch_id=${branch_id}" >> $GITHUB_OUTPUT
-        echo "db_url=postgres://${PGUSERNAME}:${PGPASSWORD}@${endpoint_id}.${region}.${cloud_provider}.neon.tech" >> $GITHUB_OUTPUT
-        echo "db_url_with_pooler=postgres://${PGUSERNAME}:${PGPASSWORD}@${endpoint_id}-pooler.${region}.${cloud_provider}.neon.tech" >> $GITHUB_OUTPUT
-
+        echo "db_url=$(yarn -s neonctl cs --project.id ${{ inputs.project_id }} --endpoint.id $endpoint_id --role.name ${{ inputs.username }} --database.name ${{ inputs.database }} --prisma ${{ inputs.prisma }})" >> $GITHUB_OUTPUT
+        echo "db_url_with_pooler=$(yarn -s neonctl cs --project.id ${{ inputs.project_id }} --endpoint.id $endpoint_id --role.name ${{ inputs.username }} --database.name ${{ inputs.database }} --pooled --prisma ${{ inputs.prisma }})" >> $GITHUB_OUTPUT
```

Comment:
The DB URL changes aren't backwards compatible. I've been appending the `/db` manually in a later step but now it ends up being `/neondb/db` 🤷

Reply:
Sorry for that! We plan to merge this to the main and publish v1 version, which will never experience incompatible updates.